src/framework/components/anim/component.js

```diff
@@ -447,17 +447,16 @@ class AnimComponent extends Component {
         this._layerIndices = {};
         this._parameters = {};
         this._playing = false;
-        Object.keys(this._targets).forEach((targetKey) => {
-            this._targets[targetKey].unbind();
-        });
+        this.unbind();
         // clear all targets from previous binding
         this._targets = {};
     }
 
     resetStateGraph() {
+        const assetId = this._stateGraphAsset;
         this.removeStateGraph();
-        if (this.stateGraphAsset) {
-            const stateGraph = this.system.app.assets.get(this.stateGraphAsset).resource;
+        if (assetId) {
+            const stateGraph = this.system.app.assets.get(assetId).resource;
             this.loadStateGraph(stateGraph);
```

Comment:
`_stateGraphAsset` is still null after this call. Should we set this prior to the call?